spaceship/integration/example_matcher.rb

```diff
@@ -0,0 +1,90 @@
+require 'diff_matcher'
+require 'multi_json'
+
+RSpec::Matchers.define :match_apple_ten_char_id do |example_path|
+  match do |actual|
+    @opts = { color_enabled: RSpec.configuration.color_enabled? }
+    @difference = DiffMatcher::Difference.new(/^[a-zA-Z0-9]{10}$/, actual, @opts)
+    @difference.matching?
+  end
+
+  failure_message do |model|
+    @difference.to_s
+  end
+end
+
+RSpec::Matchers.define :match_a_udid do |example_path|
+  match do |actual|
+    @opts = { color_enabled: RSpec.configuration.color_enabled? }
+    @difference = DiffMatcher::Difference.new(/^[a-fA-F0-9]{40}$/, actual, @opts)
+    @difference.matching?
+  end
+
+  failure_message do |model|
+    @difference.to_s
+  end
+end
+
+EXAMPLE_MATCHERS = {
+  'number' => ->(arg) { arg.kind_of? Fixnum },
+  'anything-or-empty' => ->(arg) { /.*/ || arg.nil? },
+  'apple-app-id' => /^[A-Z0-9]{10}$/,
+  'boolean' => ->(arg) { arg.kind_of?(TrueClass) || arg.kind_of?(FalseClass) },
+  'email' => /^[^@]+@\w+.\w+$/,
+  'text' => /.+/,
+  'anything' => /.*/,
+  'url' => %r{http[s]?://.*}
+}.freeze
+
+RSpec::Matchers.define :match_example do |example_path|
```

Comment:
Agreed. This is some code that @i2amsam reused from integration tests for the Crashlytics Beta Rails app. If we decide to go forward with using it, we'll document it better.

I'm curious about whether you see what's going on there and whether you like it?